examples/poisson-solver/poisson-solver.cpp

```diff
@@ -69,9 +69,9 @@ void generate_rhs(Closure f, ValueType u0, ValueType u1,
 {
     const auto discretization_points = rhs->get_size()[0];
     auto values = rhs->get_values();
-    const ValueType h = 1.0 / (discretization_points + 1);
+    const ValueType h = 1.0 / static_cast<ValueType>(discretization_points + 1);
     for (gko::size_type i = 0; i < discretization_points; ++i) {
-        const auto xi = ValueType(i + 1) * h;
```

Comment:
They don't like constructor style cast?

Reply:
They have no issue with this, but I thought since we already moved to `static_cast` mostly everywhere, I could do the same here. Not really a comprehensive change, though.